sources/pciclient/src/private.rs

```diff
@@ -148,6 +158,20 @@ pub(crate) fn check_efa_attachment<T: CommandExecutor>(command_executor: T) -> R
         .any(|device_info| device_info.device.contains(EFA_KEYWORD)))
 }
 
+/// Call `lspci` and check if there is any Neuron device attached.
+/// Internal usage, adding command_executor as parameter allows us to better unit test.
+/// For external usage, check [`crate::is_neuron_attached`].
+pub(crate) fn check_neuron_attachment<T: CommandExecutor>(command_executor: T) -> Result<bool> {
+    let neuron_devices: HashSet<_> = NEURON_DEVICES.iter().map(|s| s.to_string()).collect();
```

Comment:
Nit: can we use lazy_static and make this HastSet static? Althought, It is likely not much of a performance improvement given the size of the array and the number of times we invoke this.